### PR TITLE
Updating to use Dotnet-Cli build

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @victorhurdugaci \ @dougbu

**Comment by pranavkm** [comment]
cc @NTaylorMullen

**Comment by victorhurdugaci** [comment]
Looks good. If the checks pass :shipit:

**Comment by NTaylorMullen** [comment]
:shipit:

File: .travis.yml [hunk 1]
```diff
@@ -10,14 +10,11 @@ addons:
     - libssl-dev
     - libunwind8
     - zlib1g
-before_install:
-  - if test "$TRAVIS_OS_NAME" == "osx"; then brew update; brew install icu4c; fi
```
Comment (dougbu) [inline .travis.yml]: ICU should still be required. Do builds on OSx succeed after this removal? Either way, we've been consistent in listing all requirements whether or not the current (Beta) Travis system includes it by default. See above `addons` list.

File: .travis.yml [hunk 2]
```diff
@@ -10,14 +10,11 @@ addons:
...
-  - KOREBUILD_DNU_RESTORE_CORECLR=true MONO_THREADS_PER_CPU=2000 KOREBUILD_TEST_DNXCORE=true
 mono:
   - 4.0.5
 os:
   - linux
   - osx
+osx_image: xcode7.1
 script:
-  - ./build.sh --quiet verify
+  - ./build.sh verify
```
Comment (dougbu) [inline .travis.yml]: Why? Especially since the templates use `--quiet` and DorkScripts support that option.

File: appveyor.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 init:
   - git config --global core.autocrlf true
 build_script:
-  - build.cmd --quiet verify
+  - build.cmd verify
```
Comment (dougbu) [inline appveyor.yml]: Again, leave the build `--quiet` and somewhat readable.

File: test/Microsoft.AspNet.Razor.Runtime.Test/project.json [hunk 4]
```diff
@@ -12,19 +12,27 @@
...
     "Microsoft.AspNet.Testing": "1.0.0-*",
     "Microsoft.Extensions.WebEncoders": "1.0.0-*",
-    "xunit.runner.aspnet": "2.0.0-aspnet-*"
+    "xunit": "2.1.0"
   },
+  "testRunner": "xunit",
   "commands": {
     "test": "xunit.runner.aspnet"
   },
+  "resource": "TestFiles/**",
```
Comment (dougbu) [inline test/Microsoft.AspNet.Razor.Runtime.Test/project.json]: Does the `XmlDocumentationProvider` somehow read the XML file from a resource? If not, why make this change?

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNet.Razor.Runtime.Test/project.json (hunk 4 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Razor.Runtime.Test/project.json]: Oops, need to remove this.

File: appveyor.yml (hunk 3 above)
Comment (pranavkm) [inline appveyor.yml]: Sure.

File: .travis.yml (hunk 2 above)
Comment (pranavkm) [inline .travis.yml]: I picked up older templates. Will update.

File: .travis.yml (hunk 1 above)
Comment (pranavkm) [inline .travis.yml]: The build succeeds without this - https://travis-ci.org/aspnet/Razor/builds/101934136. Why do we need it in this repo in particular and not others?

**Comment by pranavkm** [comment]
Updated

File: .travis.yml (hunk 1 above)
Comment (dougbu) [inline .travis.yml]: Should have been in all repos. That should have been the state wherever we had Travis builds on OSx. On the other hand, previously didn't have Travis builds on OSx in most repos.

Comment (pranavkm) [inline .travis.yml]: cc @victorhurdugaci - would you know why this works without installing icu? The template doesn't have this and other repos have been succeeding.

Comment (pranavkm) [inline .travis.yml]: Spoke to @dougbu offline. Leaving the flags in for now.

File: .gitignore [hunk 5]
```diff
@@ -2,10 +2,13 @@
 [Bb]in/
 TestResults/
 .nuget/
+.build/
```
Comment (dougbu) [inline .gitignore]: one

File: .gitignore [hunk 6]
```diff
@@ -2,10 +2,13 @@
...
+.testPublish/
 *.sln.ide/
 _ReSharper.*/
 packages/
 artifacts/
+.build/
```
Comment (dougbu) [inline .gitignore]: two

File: .gitignore [hunk 7]
```diff
@@ -24,5 +28,4 @@ nuget.exe
 *.ncrunchsolution
 *.*sdf
 *.ipch
-project.lock.json
-.vs
\ No newline at end of file
+.build/
```
Comment (dougbu) [inline .gitignore]: three

**Comment by dougbu** [comment]
:shipit:

File: .gitignore (hunk 5 above)
Comment (pranavkm) [inline .gitignore]: File a bug - https://github.com/aspnet/Universe/blob/dev/build-template-dotnet/.gitignore#L31 :)

**Pull request closed without merging**